rfcs/libraries.md

```diff
@@ -0,0 +1,121 @@
+# Libraries
+
+## Summary
+
+We need a way to group together related code and provide easy ways to require them.
+
+## Motivation
+
+- Package management
+- Third party libraries
+- Simpler requires
+
+## Design
+
+### String Syntax
+
+Our require syntax should:
+
+- Work the same way in the Roblox engine and Luau language
+- Be resolvable at compile-time
+
+#### Absolute
+
+Modules can be required by their absolute path from the root of the filesystem.
+
+Requiring a file called `MyModule.luau` in `C:/MyLibrary`:
+```lua=
+local MyModule = require("C:/MyLibrary/MyModule")
```

Comment:
Will we allow `\` separators in paths?  Do we just map `/` to `\` on Windows?

Reply:
The intention was to map `/` to `\` on Windows but it seems reasonable to support `\` as well

Reply:
I have clarified this under Platforms